Update cats-effect to 1.0.0-RC2

wemrysi opened the pull request:
Updates `cats-effect` to `1.0.0-RC2` and adds missing implementations.

There are 3 laws that fail intermittently
* `runSyncStepRunAsyncConsistency`
* `toIORunAsyncConsistency`
* `toIOStackSafety`

I haven't been able to determine why. I've tried quite a few variations of implementations of `runAsync`/`runSyncStep`/`toIO` to no avail. I'm guessing one of them isn't quite correct, any ideas @ChristopherDavenport?

wemrysi committed:
Update cats-effect to 1.0.0-RC2

ChristopherDavenport commented:
Wow, these intermittent failures are frustrating. I'm not sure if the problems are the test or in scalaz.

wemrysi commented:
Same, I haven't yet been able to determine where the issue lies.

wemrysi committed:
Avoid the 'Task.apply' constructor in Arbitrary[Task]

'apply' closes over an 'ExecutorService' and runs the async callback on
that pool which introduces nondeterminism in the tests due to the Eq[IO[A]]
instance requiring the 'TestContext' to be the _only_ pool to which
tasks are submitted. h/t to @djspiewak for the fix.

wemrysi commented:
@ChristopherDavenport Thanks to @djspiewak, we have a fix! Turns out the problem was due to the use of the `Task.apply` constructor in the `Arbitrary[Task]` instance. `apply` closes over an `ExecutorService` and runs the async callback on that pool which introduces nondeterminism in the tests due to the `Eq[IO[A]]` instance requiring the `TestContext` to be the _only_ pool to which tasks are submitted.

ChristopherDavenport reviewed (approved):
Yay! For legacy compat we can release and advise use of shims-effect.

ChristopherDavenport commented:
Closed in favor of #34
